winch/codegen/src/isa/x64/masm.rs

```diff
@@ -369,6 +369,65 @@ impl Masm for MacroAssembler {
         }
     }
 
+    fn float_add(&mut self, dst: Reg, lhs: Reg, rhs: Reg, size: OperandSize) {
+        Self::ensure_two_argument_form(&dst, &lhs);
+        self.asm.xmm_add_rr(rhs, dst, size);
+    }
+
+    fn float_sub(&mut self, dst: Reg, lhs: Reg, rhs: Reg, size: OperandSize) {
+        Self::ensure_two_argument_form(&dst, &lhs);
+        self.asm.xmm_sub_rr(rhs, dst, size);
+    }
+
+    fn float_mul(&mut self, dst: Reg, lhs: Reg, rhs: Reg, size: OperandSize) {
+        Self::ensure_two_argument_form(&dst, &lhs);
+        self.asm.xmm_mul_rr(rhs, dst, size);
+    }
+
+    fn float_div(&mut self, dst: Reg, lhs: Reg, rhs: Reg, size: OperandSize) {
+        Self::ensure_two_argument_form(&dst, &lhs);
+        self.asm.xmm_div_rr(rhs, dst, size);
+    }
+
+    fn float_min(&mut self, dst: Reg, lhs: Reg, rhs: Reg, size: OperandSize) {
+        Self::ensure_two_argument_form(&dst, &lhs);
+        self.asm.xmm_min_max_seq(rhs, dst, true, size);
+    }
+
+    fn float_max(&mut self, dst: Reg, lhs: Reg, rhs: Reg, size: OperandSize) {
+        Self::ensure_two_argument_form(&dst, &lhs);
+        self.asm.xmm_min_max_seq(rhs, dst, false, size);
+    }
+
+    fn float_copysign(&mut self, dst: Reg, lhs: Reg, rhs: Reg, size: OperandSize) {
+        Self::ensure_two_argument_form(&dst, &lhs);
+        let scratch_gpr = regs::scratch();
+        let scratch_xmm = regs::scratch_xmm();
+        let clear_sign_mask = match size {
+            OperandSize::S32 => I::I32(0x7FFFFFFF),
+            OperandSize::S64 => I::I64(0x7FFFFFFFFFFFFFFF),
+            OperandSize::S128 => unreachable!(),
+        };
+        let sign_mask = match size {
+            OperandSize::S32 => I::I32(0x80000000),
+            OperandSize::S64 => I::I64(0x8000000000000000),
+            OperandSize::S128 => unreachable!(),
+        };
+
+        // Clear sign bit in dst.
+        self.load_constant(&clear_sign_mask, scratch_gpr, size);
+        self.asm.gpr_to_xmm(scratch_gpr, scratch_xmm, size);
+        self.asm.xmm_and_rr(scratch_xmm, dst, size);
```

Comment:
I think we should be able to reduce the amount of instructions by [using `andnps`,](https://www.felixcloutier.com/x86/andnps) instead of emitting an extra load constant for clearing the sign bit, so this would become:

```
andnps signbit, a
andps signbit, b
xorps ...
```